packages/server/src/auth/types/request.ts

```
@@ -18,3 +18,50 @@ export function isNodeResponse(response: GenericResponse): response is ServerRes
 export function isFetchResponse(response: GenericResponse): response is Response {
     return "headers" in response && typeof response.headers.append === "function";
 }
+
+export function getNodeRequestBody(req: IncomingMessage) {
+    // For express.js
+    if ("body" in req) {
+        return Promise.resolve(req.body);
+    }
+
+    return new Promise((resolve, reject) => {
+        let body = "";
+
+        req.on("data", (chunk) => {
+            body += chunk.toString();
+        });
+
+        req.on("end", () => {
+            try {
+                resolve(JSON.parse(body));
+            } catch (_) {
+                resolve(body);
```

I would add a comment that explains why this error is ignored.

```
// If it fails to parse the body as JSON, just return the body as a string.
```